vm/nova/vm.go

```diff
@@ -623,7 +628,12 @@ func accountDestructionValid(vmParams *vm.Params, input *vm.ChainOutputWithIDs)
 		if blockIssuerFeat.ExpirySlot >= vmParams.WorkingSet.Commitment.Slot {
 			return ierrors.Wrap(iotago.ErrInvalidBlockIssuerTransition, "cannot destroy output until the block issuer feature expires")
 		}
-		if bic, exists := vmParams.WorkingSet.BIC[outputToDestroy.AccountID]; exists {
+		accountID, is := input.ChainID.(iotago.AccountID)
+		if !is {
+			return ierrors.Wrapf(iotago.ErrBlockIssuanceCreditInputRequired, "cannot convert chain ID to account ID")
+		}
+
+		if bic, exists := vmParams.WorkingSet.BIC[accountID]; exists {
 			if bic < 0 {
 				return ierrors.Wrap(iotago.ErrAccountLocked, "cannot destroy locked account")
 			}
```

Comment:
Same here, should we add a helper function for this btw?

Reply:
Not sure what helper function you actually mean, but I think it's fine to just include the chain id in the error. Should never happen anyway because we're already in the account code path. If it fails, it's a bug in the VM.